examples/dreambooth/train_dreambooth_lora.py

```diff
@@ -473,13 +491,17 @@ def __getitem__(self, index):
         if not instance_image.mode == "RGB":
             instance_image = instance_image.convert("RGB")
         example["instance_images"] = self.image_transforms(instance_image)
-        example["instance_prompt_ids"] = self.tokenizer(
-            self.instance_prompt,
-            truncation=True,
-            padding="max_length",
-            max_length=self.tokenizer.model_max_length,
-            return_tensors="pt",
-        ).input_ids
+
+        if self.encoder_hidden_states is not None:
+            example["instance_prompt_ids"] = self.encoder_hidden_states
+        else:
+            example["instance_prompt_ids"] = self.tokenizer(
+                self.instance_prompt,
+                truncation=True,
+                padding="max_length",
+                max_length=self.tokenizer.model_max_length,
```

Comment:
I think T5 also cuts of at 77 (see: https://github.com/huggingface/diffusers/blob/0ffac97933d8ca1487d4ae9c374504a0979f0c5d/src/diffusers/pipelines/deepfloyd_if/pipeline_if.py#L298) no while `self.tokenizer.model_max_length` is much larger (512) 
Should we maybe add a `--tokenizer_max_length` arg?

Reply:
Yes, this is a good point

Reply:
Ok, I refactored the text encoding in this commit, we now have two flags for the max length and for if to use the attention mask or not. So we can match both the tokenization for stable diffusion and IF via cli

https://github.com/huggingface/diffusers/pull/3360/commits/4042274a1a3df6c587fc160eacb7f4f8058a4c8a